### PR TITLE
feat(dropdown): make updateByValue the default

### PR DESCRIPTION
**Pull request by kumarivijeta**
## Summary

The updateByValue prop has been removed, as it is now the default behavior.

## ADO Story 

[USER STORY 1931299](https://dev.azure.com/Kyndryl/Shidoka%20-%20Bridge%20Design%20System/_workitems/edit/1931299)

## Checklist

- [x] Used Conventional Commit messages as outlined in the contributing guide.
  - [x] Noted breaking changes (if any).
- [x] Documented/updated all props, events, slots, parts, etc with JSDoc.
  - [x] Ran the `analyze` command to update Storybook docs.
- [x] Added/updated Stories with controls to cover all variants.
- [ ] Ran `test` locally to address any failures.
- [ ] Added/updated the Figma link for the Story's Design tab.
- [ ] Added any new component exports to the src/index.ts file

### CODE REVIEW

**Comment by brian-patrick-3** [comment]
Oh I thought of one more important thing. We should make the `selected` prop on dropdown-option private/internal. Maybe rename it to `_selected` also.

File: src/components/reusable/dropdown/dropdown.stories.js
Comment (brian-patrick-3) [inline src/components/reusable/dropdown/dropdown.stories.js:63]: `value` should be a control/arg in each of these stories

File: src/components/reusable/dropdown/dropdown.stories.js
Comment (brian-patrick-3) [reply src/components/reusable/dropdown/dropdown.stories.js:282]: This is still pending.

**Review by brian-patrick-3 — CHANGES_REQUESTED** [review]

**Review by brian-patrick-3 — APPROVED** [review]
LGTM! Once QA approves, you can squash merge and add the `BREAKING CHANGE:` footer notes.

**Review by srpriyankashetty — APPROVED** [review]
Verified Single, Single searchable, Multiple searchable, Multi select, Grouped and Data driven Option dropdown, by adding the value and looks fine.

image

**Pull request merged**
